src/server.rs

```diff
@@ -37,7 +38,8 @@ pub struct Server {
     #[serde(skip)]
     player_records: PlayerRecords,
     #[serde(skip)]
-    friends_list: Vec<Friend>,
+    friends_lists: HashMap<SteamID, Vec<Friend>>,
+    friends_is_public: HashMap<SteamID, bool>, // True if public, False if private, no entry if we haven't checked yet.
```

Comment:
I think ideally we would store the friends lists in the actual Player objects to keep it all contained, and also allow old and unused data to flush out when they disconnect as opposed to accumulating over time. 

To avoid re-fetching data all the time on players who leave and rejoin we could check in the player history when they join the server, but that will probably be best left for another MR.